### ISSUE TITLE
Better heuristic for high entropy cookies

### ISSUE THREAD

**Issue by cooperq** [EFForg/privacybadger#771]
I think that there is a better heuristic possible for estimating cookie entropy, it probably looks something like this: 

``` javascript
var str = '103affFfffff';
var maxPossibleSymbols = 0;
var maxCombinations = 0;
var maxBits = 0;

if(/[a-z]+/.test(str)){ maxPossibleSymbols += 26; } //lower alpha
if(/[A-Z]+/.test(str)){ maxPossibleSymbols += 26; } //upper alpha
if(/[0-9]+/.test(str)){ maxPossibleSymbols += 10; } //numeric
if(/[_\W]/.test(str)){ maxPossibleSymbols += 32; } // punctuation
if(str.match(/^[a-f0-9]+$/i)){ maxPossibleSymbols = 16; } // hex
if(str.match(/^[01]+$/i)){ maxPossibleSymbols = 2; } // binary
if(str.match(/^true|false|opt-out|outout|opt_out|nodata|no_data|no-data|yes|no$/i)){ maxPossibleSymbols = 2;
maxCombinations=2;} // binary
if(!maxCombinations){
    maxCombinations = Math.pow(maxPossibleSymbols, str.length);
}
maxEntropy = Math.log(maxCombinations)/Math.LN2;
console.log(maxPossibleSymbols, maxCombinations, maxEntropy)
return maxEntropy;
```

**Comment by pde** [comment]
This is probably better than what we have, but I'm not sure that chasing better entropy estimations is all that valuable. Probably the next interesting step would be to get samples of the cookies out there that are close to the line as we currently draw it, and see how common it is for this algorithm to be making important decisions.

**Comment by cooperq** [comment]
Depends on #1019

**Comment by ghostwords** [comment]
Closing in favor of #2114.
